app/controllers/AddBusinessTradeController.scala

```diff
@@ -0,0 +1,115 @@
+                                          (implicit val appConfig: FrontendAppConfig,
+                                             implicit val itvcErrorHandlerAgent: AgentItvcErrorHandler,
+                                             implicit override val mcc: MessagesControllerComponents,
+                                             val ec: ExecutionContext)
+  extends ClientConfirmedController with I18nSupport with FeatureSwitching{
+
+  def show: Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit user =>
+      handleRequest(isAgent = false)
+  }
+
+  def showAgent: Action[AnyContent] =
+    Authenticated.async {
+      implicit request =>
+        implicit user =>
+          getMtdItUserWithIncomeSources(incomeSourceDetailsService, useCache = true) flatMap {
+            implicit mtdItUser =>
+              handleRequest(isAgent = true)
+          }
+    }
+
+  def handleRequest(isAgent: Boolean)(implicit user: MtdItUser[_], ec: ExecutionContext): Future[Result] = {
+    Future {
+      if (isDisabled(IncomeSources)) {
+        Redirect(controllers.routes.HomeController.show())
+      } else {
+        if(!isAgent) Ok(addBusinessTradeView(BusinessTradeForm.form, routes.AddBusinessTradeController.submit(), isAgent))
+        else Ok(addBusinessTradeView(BusinessTradeForm.form, routes.AddBusinessTradeController.agentSubmit(), isAgent))
+      }
+    }
+  }
+
+  def submit: Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit request =>
+      BusinessTradeForm.form.bindFromRequest().fold(
+        formWithErrors => {
+          Future {
+            Ok(addBusinessTradeView(formWithErrors, routes.AddBusinessTradeController.submit(), false))
+          }
+        },
+        formData => {
+          //if (formData.trade == request.session.get("addBusinessName").get)
+          Future {
+            Redirect("/report-quarterly/income-and-expenses/view/income-sources/add/business-address").withSession(request.session + ("addBusinessTrade" -> formData.trade))
+          }
+        }
+      )
+    }
+
+  def agentSubmit: Action[AnyContent] = Authenticated.async {
+    implicit request =>
+      implicit user =>
+        getMtdItUserWithIncomeSources(incomeSourceDetailsService, useCache = true) flatMap {
+          implicit mtdItUser =>
+            BusinessTradeForm.form.bindFromRequest().fold(
+              formWithErrors => {
+                Future {
+                  Ok(addBusinessTradeView(formWithErrors, routes.AddBusinessTradeController.agentSubmit(), true))
+                }
+              },
+              formData => {
+                //if (formData.trade == request.session.get("addBusinessName").get)
+                Future {
```

Comment:
if you can use Future.success instead, this would generated already completed Future etc;